### PR TITLE
[test-api-update] dagster-graphql

### PR DESCRIPTION
**Pull request by smackesey**
## Summary & Motivation

Migrate dagster_graphql_tests fully over to job/op/graph APIs.

## How I Tested These Changes

BK

### CODE REVIEW

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #13145** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13145" target="_blank">Graphite</a> 
    * **PR #13108** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13108" target="_blank">Graphite</a> 
      * **PR #13168** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13168" target="_blank">Graphite</a> 
        * **PR #13169** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13169" target="_blank">Graphite</a> 
          * **PR #13081** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13081" target="_blank">Graphite</a> 
            * **PR #13083** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13083" target="_blank">Graphite</a> 
            * **PR #12770** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12770" target="_blank">Graphite</a> 
              * **PR #12769** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12769" target="_blank">Graphite</a> 
              * **PR #13170** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13170" target="_blank">Graphite</a> 
                * **PR #13171** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13171" target="_blank">Graphite</a> 
                  * **PR #13188** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13188" target="_blank">Graphite</a> 
                    * **PR #13172** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13172" target="_blank">Graphite</a> 
                      * **PR #13192** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13192" target="_blank">Graphite</a> 
                        * **PR #13301** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13301" target="_blank">Graphite</a> 
                          * **PR #13302** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13302" target="_blank">Graphite</a>  👈
                        * **PR #13299** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13299" target="_blank">Graphite</a> 
                        * **PR #13298** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13298" target="_blank">Graphite</a> 
                        * **PR #13297** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13297" target="_blank">Graphite</a> 
                        * **PR #13296** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13296" target="_blank">Graphite</a> 
                        * **PR #13295** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13295" target="_blank">Graphite</a> 
                        * **PR #13294** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13294" target="_blank">Graphite</a> 
                      * **PR #13173** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13173" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/13302?utm_source=stack-comment).

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #13145** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13145" target="_blank">Graphite</a> 
    * **PR #13108** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13108" target="_blank">Graphite</a> 
      * **PR #13168** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13168" target="_blank">Graphite</a> 
        * **PR #13169** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13169" target="_blank">Graphite</a> 
          * **PR #13081** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13081" target="_blank">Graphite</a> 
            * **PR #13083** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13083" target="_blank">Graphite</a> 
            * **PR #12770** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12770" target="_blank">Graphite</a> 
              * **PR #12769** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12769" target="_blank">Graphite</a> 
              * **PR #13170** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13170" target="_blank">Graphite</a> 
                * **PR #13171** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13171" target="_blank">Graphite</a> 
                  * **PR #13188** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13188" target="_blank">Graphite</a> 
                    * **PR #13172** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13172" target="_blank">Graphite</a> 
                      * **PR #13192** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13192" target="_blank">Graphite</a> 
                        * **PR #13303** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13303" target="_blank">Graphite</a> 
                        * **PR #13301** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13301" target="_blank">Graphite</a> 
                          * **PR #13302** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13302" target="_blank">Graphite</a>  👈
                        * **PR #13299** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13299" target="_blank">Graphite</a> 
                        * **PR #13298** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13298" target="_blank">Graphite</a> 
                        * **PR #13297** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13297" target="_blank">Graphite</a> 
                        * **PR #13296** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13296" target="_blank">Graphite</a> 
                        * **PR #13295** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13295" target="_blank">Graphite</a> 
                        * **PR #13294** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13294" target="_blank">Graphite</a> 
                      * **PR #13173** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13173" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/13302?utm_source=stack-comment).

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #13145** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13145" target="_blank">Graphite</a> 
    * **PR #13108** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13108" target="_blank">Graphite</a> 
      * **PR #13168** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13168" target="_blank">Graphite</a> 
        * **PR #13169** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13169" target="_blank">Graphite</a> 
          * **PR #13081** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13081" target="_blank">Graphite</a> 
            * **PR #13083** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13083" target="_blank">Graphite</a> 
            * **PR #12770** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12770" target="_blank">Graphite</a> 
              * **PR #12769** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12769" target="_blank">Graphite</a> 
              * **PR #13170** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13170" target="_blank">Graphite</a> 
                * **PR #13171** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13171" target="_blank">Graphite</a> 
                  * **PR #13188** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13188" target="_blank">Graphite</a> 
                    * **PR #13172** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13172" target="_blank">Graphite</a> 
                      * **PR #13192** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13192" target="_blank">Graphite</a> 
                        * **PR #13303** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13303" target="_blank">Graphite</a> 
                        * **PR #13301** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13301" target="_blank">Graphite</a> 
                          * **PR #13302** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13302" target="_blank">Graphite</a>  👈
                        * **PR #13299** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13299" target="_blank">Graphite</a> 
                        * **PR #13298** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13298" target="_blank">Graphite</a> 
                        * **PR #13297** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13297" target="_blank">Graphite</a> 
                        * **PR #13296** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13296" target="_blank">Graphite</a> 
                        * **PR #13295** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13295" target="_blank">Graphite</a> 
                        * **PR #13294** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13294" target="_blank">Graphite</a> 
                      * **PR #13173** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13173" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/13302?utm_source=stack-comment).

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #13145** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13145" target="_blank">Graphite</a> 
    * **PR #13108** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13108" target="_blank">Graphite</a> 
      * **PR #13168** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13168" target="_blank">Graphite</a> 
        * **PR #13169** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13169" target="_blank">Graphite</a> 
          * **PR #13081** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13081" target="_blank">Graphite</a> 
            * **PR #13083** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13083" target="_blank">Graphite</a> 
            * **PR #12770** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12770" target="_blank">Graphite</a> 
              * **PR #12769** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12769" target="_blank">Graphite</a> 
              * **PR #13170** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13170" target="_blank">Graphite</a> 
                * **PR #13171** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13171" target="_blank">Graphite</a> 
                  * **PR #13188** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13188" target="_blank">Graphite</a> 
                    * **PR #13172** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13172" target="_blank">Graphite</a> 
                      * **PR #13192** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13192" target="_blank">Graphite</a> 
                        * **PR #13303** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13303" target="_blank">Graphite</a> 
                        * **PR #13301** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13301" target="_blank">Graphite</a> 
                          * **PR #13302** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13302" target="_blank">Graphite</a>  👈
                        * **PR #13299** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13299" target="_blank">Graphite</a> 
                        * **PR #13298** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13298" target="_blank">Graphite</a> 
                        * **PR #13297** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13297" target="_blank">Graphite</a> 
                        * **PR #13296** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13296" target="_blank">Graphite</a> 
                        * **PR #13295** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13295" target="_blank">Graphite</a> 
                        * **PR #13294** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13294" target="_blank">Graphite</a> 
                      * **PR #13173** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13173" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/13302?utm_source=stack-comment).

File: python_modules/dagster-graphql/dagster_graphql_tests/client_tests/sleepy.py [hunk 1]
```diff
@@ -54,23 +51,10 @@ def total(_, in_1, in_2, in_3, in_4):
...
-            "multi",
-            {
-                "execution": {"multiprocess": {}},
-                "solids": {"giver": {"config": [2, 2, 2, 2]}},
-            },
-        )
-    ],
-    mode_defs=[default_mode_def_for_test],
+@job(
+    description="Demo diamond-shaped job that has four-path parallel structure of ops.",
```
Comment (dpeng817) [inline python_modules/dagster-graphql/dagster_graphql_tests/client_tests/sleepy.py:55]: should maybe specify a config arg for that op config?

File: python_modules/dagster-graphql/dagster_graphql_tests/client_tests/test_submit_pipeline_execution.py [hunk 2]
```diff
@@ -189,7 +172,8 @@ def test_no_location_or_repo_provided_success(mock_client: MockClient):
     ]
 
     actual_run_id = mock_client.python_client.submit_pipeline_execution(
-        pipeline_name, run_config={}, mode="default"
+        pipeline_name,
+        run_config={},
```
Comment (dpeng817) [inline python_modules/dagster-graphql/dagster_graphql_tests/client_tests/test_submit_pipeline_execution.py:176]: unrelated to this PR, but rough that this is necessary

File: python_modules/dagster-graphql/dagster_graphql_tests/client_tests/sleepy.py (hunk 1 above)
Comment (smackesey) [reply python_modules/dagster-graphql/dagster_graphql_tests/client_tests/sleepy.py:55]: ok

**Review by dpeng817 — APPROVED** [review]

**Pull request merged**